frontend/src/UnitCounter.tsx

```diff
@@ -21,8 +21,20 @@ export function UnitCounter({
     onChange,
 }: Props) {
     return (
-        <Stack direction="row" spacing={1}>
-            <Box component="img" alt={unit} src={imgPath} />
+        <Stack direction="row" spacing={1} alignItems="center">
+            <Typography
+                level="body-xs"
+                startDecorator={
+                    <Box
+                        component="img"
+                        alt={unit}
+                        src={imgPath}
+                        maxWidth={44}
```

Comment:
This reminds me that we need to test this on a mobile browser, since that's how everyone will actually be using it. Do we want to do some sort of percentage-of-screen-width thing here, instead of a fixed value?

Reply:
Totally right, I ended up making a few changes for this, because it was hard to address it in isolation!

- Customized MUI's breakpoints a little bit

- For unit icons, added a `minWidth` of 20 and kept the `maxWidth` of 44 (rather than a % of screen width or % of the containing div, this will allow the image to resize between 20px (any smaller is too hard to see) and 44px (any higher and the infantry icon would stay tiny while the rest get big, looks weird)

- For the unit text label, added a fixed width of 80px, but when we hit our custom `sm` breakpoint, the text labels will just disappear to give us the space back

- For the unit cost and counter inputs, set their widths relative to the font size, so they should always stay at least large enough to display the contained font with some padding

https://github.com/user-attachments/assets/1a8a4913-5212-46af-a6b2-a939b85e4d11